src/pgroonga.c

```diff
@@ -5391,11 +5391,20 @@ PGrnSearchIsInCondition(ScanKey key)
 			key->sk_strategy == PGrnEqualStrategyNumber);
 }
 
+static bool
+PGrnSearchIsMatchInCondition(ScanKey key)
+{
+	return (key->sk_flags & SK_SEARCHARRAY) &&
+		   ((key->sk_strategy == PGrnMatchStrategyNumber) ||
+			(key->sk_strategy == PGrnMatchStrategyV2Number));
+}
+
 static void
 PGrnSearchBuildConditionIn(PGrnSearchData *data,
 						   ScanKey key,
 						   grn_obj *targetColumn,
-						   Form_pg_attribute attribute)
+						   Form_pg_attribute attribute,
+						   bool matchIn)
```

Comment:
If we want to pass something to `PGrnSearchBuildConditionIn()`, we should use `grn_operator op` instead of `bool matchIn`.

If `op` is `GRN_OP_EQUAL`, we can optimize it by `in_values()`.
We should use `column OP value` (such of `GRN_OP_MATCH`) otherwise.